findlibs 0.1.0 doesn't support Python below 3.10

### What happened?

From version `0.1.0` of `findlibs` it uses [union type expressions](https://docs.python.org/3.10/library/stdtypes.html#types-union) (only allowed from Python `3.10`) but the package classifiers allow install on Python versions below `3.10` (`3.6`, `3.7` and `3.8` are explicitly tagged).

### What are the steps to reproduce the bug?

With Pythono `3.9` try to install and use `findlibs`, it will install `0.1.0` version that is incompatible.

### Version

0.1.0

### Platform (OS and architecture)

Linux 5.15.0-131-generic #141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux

### Relevant log output

```shell
[ERROR] TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'                                                                                                                                        
Traceback (most recent call last):                                                                                                                                                                                 
  File "/var/lang/lib/python3.8/imp.py", line 234, in load_module                                                                                                                                                  
    return load_source(name, filename, file)                                                                                                                                                                       
  File "/var/lang/lib/python3.8/imp.py", line 171, in load_source                                                                                                                                                  
    module = _load(spec)                                                                                                                                                                                           
  File "<frozen importlib._bootstrap>", line 702, in _load                                                                                                                                                         
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked                                                                                                                                                
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module                                                                                                                                          
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                                                                                     
  File "/var/task/lambda_function.py", line 3, in <module>                                                                                                                                                         
    import cfgrib                                                                                                                                                                                                  
  File "/opt/python/cfgrib/__init__.py", line 20, in <module>                                                                                                                                                      
    from .cfmessage import COMPUTED_KEYS                                                                                                                                                                           
  File "/opt/python/cfgrib/cfmessage.py", line 29, in <module>                                                                                                                                                     
    from . import abc, messages                                                                                                                                                                                    
  File "/opt/python/cfgrib/messages.py", line 28, in <module>                                                                                                                                                      
    import eccodes  # type: ignore                                                                                                                                                                                 
  File "/opt/python/eccodes/__init__.py", line 13, in <module>                                                                                                                                                     
    from .eccodes import *  # noqa                                                                                                                                                                                 
  File "/opt/python/eccodes/eccodes.py", line 12, in <module>
    from gribapi import (
  File "/opt/python/gribapi/__init__.py", line 13, in <module>
    from .gribapi import *  # noqa
  File "/opt/python/gribapi/gribapi.py", line 34, in <module>
    from gribapi.errors import GribInternalError
  File "/opt/python/gribapi/errors.py", line 16, in <module>
    from .bindings import ENC, ffi, lib
  File "/opt/python/gribapi/bindings.py", line 102, in <module>
    library_path = find_binary_libs("eccodes")
  File "/opt/python/gribapi/bindings.py", line 95, in find_binary_libs
    import findlibs
  File "/opt/python/findlibs/__init__.py", line 78, in <module>
    lib_name: str, pkg_name: str, preload_deps: bool | None = None
```

### Accompanying data

_No response_

### Organisation

_No response_

--- a/findlibs/__init__.py
+++ b/findlibs/__init__.py
@@ -294,7 +294,7 @@ def find(lib_name: str, pkg_name: Union[str, None] = None) -> Union[str, None]:
     sources_filtered = (
         source_clb
         for source_clb, source_name in sources
-        if os.environ.get(f"FINDLIBS_DISABLE_{source_name}", None) != "yes"
+        if os.environ.get(f"FINDLIBS_DISABLE_{source_name}", None) not in ("yes", "1")
     )
 
     for source in sources_filtered:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,16 +19,13 @@ license = { text = "Apache License Version 2.0" }
 authors = [
   { name = "European Centre for Medium-Range Weather Forecasts (ECMWF)", email = "software@ecmwf.int" }
 ]
+requires-python = ">=3.10"
 keywords = ["tool"]
 classifiers = [
   "Development Status :: 3 - Alpha",
   "Intended Audience :: Developers",
   "License :: OSI Approved :: Apache Software License",
   "Programming Language :: Python :: 3",
-  "Programming Language :: Python :: 3.6",
-  "Programming Language :: Python :: 3.7",
-  "Programming Language :: Python :: 3.8",
-  "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",

